src/theory/bv/bv_solver_bitblast.cpp

```diff
@@ -339,14 +340,16 @@ void BVSolverBitblast::initSatSolver()
       d_satSolver.reset(prop::SatSolverFactory::createCadical(
           statisticsRegistry(),
           d_env.getResourceManager(),
-          "theory::bv::BVSolverBitblast::"));
+          "theory::bv::BVSolverBitblast::",
+          isTheoryProofProducing));
   }
-  d_cnfStream.reset(new prop::CnfStream(d_env,
-                                        d_satSolver.get(),
-                                        d_bbRegistrar.get(),
-                                        d_nullContext.get(),
-                                        prop::FormulaLitPolicy::INTERNAL,
-                                        "theory::bv::BVSolverBitblast"));
+  d_cnfStream.reset(new prop::CnfStream(
+    d_env,
+    d_satSolver.get(),
+    d_bbRegistrar.get(),
+    d_nullContext.get(),
+    isTheoryProofProducing ? prop::FormulaLitPolicy::TRACK : prop::FormulaLitPolicy::INTERNAL,
```

Comment:
```suggestion
    // If we are producing proofs for the SAT solver, we need to that all literals created in the CNF stream are tracked, which is not the case with the FormulaLitPolicy::INTERNAL
    isTheoryProofProducing ? prop::FormulaLitPolicy::TRACK : prop::FormulaLitPolicy::INTERNAL,
```

You'll need to clang-format as well.